test/integration/federation/crud_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"testing"
+
+	"github.com/pborman/uuid"
+
+	"k8s.io/kubernetes/test/integration/federation/framework"
+)
+
+// TestFederationCRUD validates create/read/update/delete operations for federated resource types.
+func TestFederationCRUD(t *testing.T) {
+	fedFixture := framework.FederationFixture{ClusterCount: 2}
+	fedFixture.Setup(t)
+	defer fedFixture.Teardown(t)
+
+	resourceFixtures := []framework.ResourceFixture{
+		&framework.SecretFixture{},
+	}
+	for _, fixture := range resourceFixtures {
+		kind := fixture.Kind()
+		t.Run(kind, func(t *testing.T) {
+			framework.SetupResourceFixture(t, fedFixture.APIFixture, fixture)
+			defer fixture.Teardown(t)
+
+			crud := framework.NewCRUDHelper(t, fixture.Adapter(), fedFixture.ClusterClients)
+			obj := fixture.NewObject(uuid.New())
+			crud.CheckLifecycle(obj)
+		})
+	}
+
+	// Validate deletion handling where orphanDependents is true or nil for a single resource type since the
+	// underlying logic is common across all types.
+	orphanedDependents := true
```

Comment:
The tests are run as golang subtests and this is really just a dispatcher that enables a shared fixture.